### DISCUSSION TITLE
Use S3 buckets as network file systems

### DISCUSSION THREAD

**Discussion by uhurusurfa** [OpenAS2/OpenAs2App#307] [Ideas]
Rather than add AWS specific code, use an os utility to mount the S3 bucket as a network drive then you can run  OpenAS2 without any AWS specific code addtitions:
Linux:
https://github.com/s3fs-fuse/s3fs-fuse
Windows:
https://tntdrive.com/mount-amazon-s3-bucket.aspx

**Comment by igwtech** [comment]
Hi there,

Keep in mind that generally S3 cannot offer the same performance or
semantics as a local file system. More specifically:

   - random writes or appends to files require rewriting the entire object,
   optimized with multi-part upload copy
   - metadata operations such as listing directories have poor performance
   due to network latency
   - non-AWS providers may have eventual consistency so reads can
   temporarily yield stale data (AWS offers read-after-write consistency since
   Dec 2020)
   - no atomic renames of files or directories
   - no coordination between multiple clients mounting the same bucket
   - no hard links
   - inotify detects only local modifications, not external ones by other
   clients or tools

Regards, J

On Thu, Nov 24, 2022 at 6:55 AM Christopher Broderick <
***@***.***> wrote:

> Rather than add AWS specific code, use an os utility to mount the S3
> bucket as a network drive then you can run OpenAS2 without any AWS specific
> code addtitions:
> Linux:
> https://github.com/s3fs-fuse/s3fs-fuse
> Windows:
> https://tntdrive.com/mount-amazon-s3-bucket.aspx
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenAS2/OpenAs2App/discussions/307>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA2QND6U3YZTG5GVAL74M2DWJ5CRBANCNFSM6AAAAAASKHTZ2A>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>















-- 
Javier Munoz

CEO

Greicodex Software



[image: mobilePhone] +58 2127629120 <+58+2127629120>
[image: emailAddress] ***@***.***
[image: website] www.greicodex.com
[image: address] Av Francisco Solano, Centro Solano Plaza, Oficina PHA,
Caracas, Venezuela



[image: facebook] <https://www.facebook.com/greicodex>

[image: twitter] <https://twitter.com/greicodex>

[image: linkedin] <https://www.linkedin.com/company/greicodexsoftware>

[image: instagram] <https://www.instagram.com/greicodex/>
